pkg/apis/cnsoperator/cnsnodevmbatchattachment/v1alpha1/cnsnodebatchvmattachment_types.go

```diff
@@ -21,32 +21,38 @@ import (
 )
 
 // DiskMode describes the desired mode to use when attaching the volume.
+// +kubebuilder:validation:Enum=independent_persistent;persistent;independent_nonpersistent
 type DiskMode string
 
 const (
-	// By setting DiskMode to independent_persistent, a virtual machine's disk is not captured in snapshots and
+	// IndependentPersistent is the diskMode in which, a virtual machine's disk is not captured in snapshots and
 	// changes are permanently written to the disk, regardless of snapshot operations.
 	IndependentPersistent DiskMode = "independent_persistent"
-	// Changes are immediately and permanently written to the virtual disk.
+	// Persistent diskMode changes are immediately and permanently written to the virtual disk.
 	Persistent DiskMode = "persistent"
+	// Changes to virtual disk are made to a redo log and discarded at power off.
+	// It is not affected by snapshots.
+	IndependentNonPersistent = "independent_nonpersistent"
 )
 
-// The sharing mode of the virtual disk.
+// SharingMode is the sharing mode of the virtual disk.
+// +kubebuilder:validation:Enum=sharingMultiWriter;sharingNone
 type SharingMode string
 
 const (
-	// The virtual disk is shared between multiple virtual machines.
+	// SharingMultiWriter: The virtual disk is shared between multiple virtual machines.
 	SharingMultiWriter SharingMode = "sharingMultiWriter"
-	// The virtual disk is not shared.
+	// SharingNone: The virtual disk is not shared.
 	SharingNone SharingMode = "sharingNone"
 )
 
-// CnsNodeVmBatchAttachmentSpec defines the desired state of CnsNodeVmBatchAttachment
+// CnsNodeVMBatchAttachmentSpec defines the desired state of CnsNodeVMBatchAttachment
 // +k8s:openapi-gen=true
-type CnsNodeVmBatchAttachmentSpec struct {
+type CnsNodeVMBatchAttachmentSpec struct {
+	// +required
```

Comment:
It looks like the position of these annotations with respect to the comments is not consistent throughout the file.  Can we follow the convention like:
```go
// +optional

// Code Comments
myField myType
```

Notice the empty line between the `optional / required` marker and the API comment.  This results in the annotation being gone from the Godoc while still being effective.  Which cleans up the APIs nicely.